perceval/backends/core/nntp.py

```diff
@@ -277,6 +216,118 @@ def parse_article(raw_article):
             raise ParseError(cause=str(e))
         return article
 
+    def _init_client(self, from_archive=False):
+        """Init client"""
+
+        return NNTTPClient(self.host, self.archive, from_archive)
+
+    def __parse_article(self, info):
+        reader = io.BytesIO(b'\n'.join(info['lines']))
+        raw_article = reader.read().decode('utf-8', errors='surrogateescape')
+        data = self.parse_article(raw_article)
+
+        article = self.__build_article(data,
+                                       info['message_id'],
+                                       info['number'])
+
+        logger.debug("Article %s (offset: %s) parsed",
+                     article['message_id'], article['offset'])
+
+        return article
+
+    def __build_article(self, article, message_id, offset):
+        a = {k: v for k, v in article.items()}
+        a['message_id'] = message_id
+        a['offset'] = offset
+        return a
+
+
+class NNTTPClient():
+    """NNTP client
+
+    :param host: host
+    :param group: name of the group
+    :param archive: an archive to store/read fetched data
+    :param from_archive: it tells whether to write/read the archive
+    """
+
+    GROUP = "group"
+    ARTICLE = "article"
+    OVER = "over"
+
+    def __init__(self, host, archive=None, from_archive=False):
+        self.host = host
+        self.archive = archive
+        self.from_archive = from_archive
+        self.handler = nntplib.NNTP(self.host)
+
+    def fetch(self, method, args):
+        """Fetch NNTP data from the server or from the archive
+
+        :param method: the name of the command to execute
+        :param args: the arguments required by the command
+        """
+        if self.from_archive:
+            data = self.fetch_from_archive(method, args)
+        else:
+            data = self.fetch_from_remote(method, args)
+
+        return data
+
+    def fetch_from_remote(self, method, args):
+        """Fetch data from NNTP
+
+        :param method: the name of the command to execute
+        :param args: the arguments required by the command
+        """
+        if method == NNTTPClient.GROUP:
+            data = self.handler.group(args)
+        elif method == NNTTPClient.OVER:
+            data = self.handler.over(args)
+        elif method == NNTTPClient.ARTICLE:
+            data = self.fetch_article(args)
+
+        if self.archive:
+            self.archive.store(method, args, "", data)
```

Comment:
When storing values in the archive, set to `None` those parameters that will never be used not an empty string.